src/main/java/org/tools4j/hoverraft/command/InMemoryCommandLog.java

```diff
@@ -33,36 +33,13 @@
 public class InMemoryCommandLog implements CommandLog {
 
     private final DirectFactory directFactory = new AllocatingDirectFactory();
-    private final List<CommandLogEntry> commandLogEntries = new ArrayList<>();
+    private final List<LogEntry> entries = new ArrayList<>();
     private final AtomicInteger readIndex = new AtomicInteger(0);
-
-    private final LogEntry lastEntry = new LogEntry() {
-        @Override
-        public int term() {
-            readIndex(index());
-            return readTerm();
-        }
-
-        @Override
-        public long index() {
-            return commandLogEntries.size() - 1;
-        }
-
-
-        @Override
-        public LogEntry term(final int term) {
-            throw new UnsupportedOperationException("term update is not allowed");
-        }
-
-        @Override
-        public LogEntry index(final long index) {
-            throw new UnsupportedOperationException("index update is not allowed");
-        }
-    };
+    private final DirectLogEntry lastEntry = new DirectLogEntry();
```

Comment:
yea i think so -- let's discuss our CommandLog is dodgy and has some problems.

Main problem is confusion between (read) index and LogKey index